addon/mixins/ui-service-mixin.js

```diff
@@ -55,7 +56,36 @@ var ServicesMixin = Mixin.create({
           });
         }
       };
+
+      // Using postMessage as an alternative to localStorage/storageEvent
```

Comment:
Let's add some updated tests  - QA is getting pretty crazy https://civicplus.tpondemand.com/entity/179336-open-new-window-vs-modal-popup/comments/309113.

Reply:
NIce, tests!